app.rb

```diff
@@ -0,0 +1,123 @@
+  end
+
+  def start_library
+    puts 'Welcome to my School Library!'
+    until list_of_options
+      answer = gets.chomp.to_i
+      if answer == 7
+        puts 'You have selected exit! bye'
+        exit
+      end
+
+      user_input answer
+    end
+  end
+
+  def list_all_books
+    puts 'Database is empty! Add a book.' if @books.empty?
+    @books.each { |book| puts "[Book] Title: #{book.title}, Author: #{book.author}" }
+  end
+
+  def list_all_people
+    puts 'Database is empty! Add a person.' if @people.empty?
+    @people.each do |person|
+      puts "[#{person.class.name}] Name: #{person.name}, Age: #{person.age}, id: #{person.id}"
+    end
+  end
+
+  def create_person
+    print 'To create a student, press 1, to create a teacher, press 2 : '
+    option = gets.chomp
+
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Invalid input. Try again'
+    end
+  end
+
+  def create_student
+    puts 'Create a new student'
+    print 'Enter student age: '
+    age = gets.chomp.to_i
+    print 'Enter name: '
+    name = gets.chomp
+    @people.push(Student.new(@classroom, age, name))
+    puts 'Student created successfully'
+  end
+
+  def create_teacher
+    puts 'Create a new teacher'
+    print 'Enter teacher age: '
+    age = gets.chomp.to_i
+    print 'Enter teacher name: '
+    name = gets.chomp
+    print 'Enter teacher specialization: '
+    specialization = gets.chomp
+    @people.push(Teacher.new(specialization, age, name))
+    puts 'Teacher created successfully'
+  end
+
+  def create_book()
+    puts 'Create a new book'
+    print 'Enter title: '
+    title = gets.chomp
+    print 'Enter author: '
+    author = gets
+    @books.push(Book.new(title, author))
+    puts "Book #{title} created successfully."
+  end
+
+  def create_rental
+    puts 'Select which book you want to rent by entering its number'
+    @books.each_with_index { |book, index| puts "#{index}) Title: #{book.title}, Author: #{book.author}" }
+
+    book_id = gets.chomp.to_i
+
+    puts 'Select a person from the list by its number'
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+
+    person_id = gets.chomp.to_i
+
+    print 'Date: '
+    date = gets.chomp.to_s
+
+    @rentals.push(Rental.new(date, @people[person_id], @books[book_id]))
+    puts 'Rental created successfully'
+  end
+
+  def list_all_rentals
+    puts 'To see person rentals enter the person ID: '
+    @people.each do |person|
+      puts "id: #{person.id}"
+    end
+    id = gets.chomp.to_i
+    puts 'Rented Books:'
+    @rentals.each do |rental|
+      if rental.person.id == id
+        puts "Peson: #{rental.person.name}  Date: #{rental.date}, Book: '#{rental.book.title}' by #{rental.book.author}"
+      else
+        puts
+        puts 'No records where found for the given ID'
+      end

```

Comment:
- [ ] when trying to list all rentals the user receives the below error:
![Screenshot 2022-04-14 211021](https://user-images.githubusercontent.com/39240820/163461482-3c448761-6bc0-42bc-bb48-523204c69579.png)
this is because the application is not able to find a `rental.person.id`. 
The reason for this error, however, comes from line 103:
https://github.com/Lichapa/OOP-School-library/blob/cbe8973dd493da07fd89ff336139e15af590f1c6/app.rb#L103
if you were to change this line to:
```
@rentals.push(Rental.new(date, @books[book_id], @people[person_id]))
```
it will fix the error because the `initialize` method in the `Rental` class takes the arguments in this order.

Reply:
Thanks for noticing. I have made the changes and it is working as expected